packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -96,26 +96,97 @@ def test_metrics_containers(dcos_api_session):
     the statsd-emitter executor. When found, query it's /app endpoint to test that
     it's sending the statsd metrics as expected.
     """
+    # Helper func to check for non-unique CID's in a given /containers/id endpoint
+    def check_cid(registry):
+        if len(registry) <= 1:
+            return True
+
+        cid1 = registry[len(registry) - 1]
+        cid2 = registry[len(registry) - 2]
+        if cid1 != cid2:
+            raise ValueError('{} != {}'.format(cid1, cid2))
+
+        return True
+
     @retrying.retry(wait_fixed=2000, stop_max_delay=LATENCY * 1000)
     def test_containers(app_endpoints):
+
+        debug_eid = []
+
         for agent in app_endpoints:
             response = dcos_api_session.metrics.get('containers', node=agent.host)
+            assert response.status_code == 200, "got {}".format(
```

Comment:
this specific assert, if it fails, will print almost exactly what your reason string prints - I think in this case you can completely drop the second argument to assert